java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java

```diff
@@ -325,6 +340,93 @@ public static void readPrependValues(VariableRegistry registry, String prependFi
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public static void processYaml(VariableRegistry registry, File[] yamlFiles, ReaderState state) throws IOException {
+        Map<Integer, String> listOutputs = new HashMap<>();
+        Map<Integer, String> listAnalogInputs = new HashMap<>();
+        Map<Integer, String> listEventInputs = new HashMap<>();
+        Map<Integer, String> listSwitchInputs = new HashMap<>();
+        for (File yamlFile : yamlFiles) {
+            Yaml yaml = new Yaml();
+            Map<String, Object> yamlData = yaml.load(new FileReader(yamlFile));
+            List<Map<String, Object>> data = (List<Map<String, Object>>) yamlData.get("pins");
+            if (data == null) {
+                SystemOut.println("Null yaml for " + yamlFile);
+            } else {
+                SystemOut.println(data);
+                Objects.requireNonNull(data, "data");
+                for (Map<String, Object> pin : data) {
+                    if (pin.get("id") instanceof ArrayList) {
+                        for (int i = 0; i < ((ArrayList) pin.get("id")).size(); i++) {
+                            findMatchingEnum((String) ((ArrayList) pin.get("id")).get(i),
```

Comment:
is that 5 levels of if/for deep? in an ideal world I would extract some smaller methods to break this super-method into bit simpler pieces?